### ISSUE TITLE
Project countour pods are not properly spawn when I create a kind cluster with ingress controller enabled (using project contour)

### ISSUE THREAD

**Issue by odockal** [containers/podman-desktop#2163]
### Bug description

I created a cluster using kind and ingress controller feature enabled. although, If I check pods state using `kubectl get pods -A` it is pending state for `projectcontour` pods. That is on linux. 

On windows I got just one pod for on the list the controur (contour-certgen), also pending.

### Operating system

Fedora 37, Windows 10

### Version

0.14.1

### Steps to reproduce

1. Start PD, install Podman + podman machine
2. Install Kind
3. Create Kind cluster with ingress controller enabled
4. Assert: pods are created and running, use kubectl to check

### Relevant log output

```shell
$ kubectl get pods -A
NAMESPACE            NAME                                                 READY   STATUS             RESTARTS         AGE
default              httpbin-749b466b9c-jgn86                             0/1     Pending            0                21m
default              httpbin-749b466b9c-lv9gw                             0/1     Pending            0                21m
default              httpbin-749b466b9c-qcx2g                             0/1     Pending            0                21m
kube-system          coredns-787d4945fb-htm8s                             0/1     Pending            0                35m
kube-system          coredns-787d4945fb-t59th                             0/1     Pending            0                35m
kube-system          etcd-kind-cluster-control-plane                      1/1     Running            0                35m
kube-system          kindnet-zzmzv                                        0/1     ImagePullBackOff   0                35m
kube-system          kube-apiserver-kind-cluster-control-plane            1/1     Running            0                35m
kube-system          kube-controller-manager-kind-cluster-control-plane   1/1     Running            0                35m
kube-system          kube-proxy-lsm8c                                     0/1     CrashLoopBackOff   11 (4m11s ago)   35m
kube-system          kube-scheduler-kind-cluster-control-plane            1/1     Running            0                35m
local-path-storage   local-path-provisioner-75f5b54ffd-v9xdk              0/1     Pending            0                35m
projectcontour       contour-74866bdd99-b5g5n                             0/1     Pending            0                35m
projectcontour       contour-74866bdd99-x9bx5                             0/1     Pending            0                35m
projectcontour       contour-certgen-v1.24.2-kqfft                        0/1     Pending            0                35m
```
```


### Additional context

_No response_

**Comment by jeffmaury** [comment]
Looks like the whole cluster is not running properly. Can you check the pod in error (kindnet and kubeproxy). `kubectl get pod -o yml` may help

**Comment by odockal** [comment]
I tried to restart, the containers that did not go up were coredns, kube-proxy and local-path-provisioner - these are the same as before, expect kindnet. Relevant outputs:
```
$ kubectl get pod kube-proxy-4pc46 -o yaml --namespace kube-system
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2023-04-19T20:03:49Z"
  generateName: kube-proxy-
  labels:
    controller-revision-hash: 5cbfdcddd5
    k8s-app: kube-proxy
    pod-template-generation: "1"
  name: kube-proxy-4pc46
  namespace: kube-system
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: kube-proxy
    uid: ed3f67e9-d305-4aad-959c-53d113b56c05
  resourceVersion: "1164"
  uid: deed1d8c-fce3-494e-8e1f-52d7d0d2e819
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - kind-cluster-control-plane
  containers:
  - command:
    - /usr/local/bin/kube-proxy
    - --config=/var/lib/kube-proxy/config.conf
    - --hostname-override=$(NODE_NAME)
    env:
    - name: NODE_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    image: registry.k8s.io/kube-proxy:v1.26.3
    imagePullPolicy: IfNotPresent
    name: kube-proxy
    resources: {}
    securityContext:
      privileged: true
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/lib/kube-proxy
      name: kube-proxy
    - mountPath: /run/xtables.lock
      name: xtables-lock
    - mountPath: /lib/modules
      name: lib-modules
      readOnly: true
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-m9xcs
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  hostNetwork: true
  nodeName: kind-cluster-control-plane
  nodeSelector:
    kubernetes.io/os: linux
  preemptionPolicy: PreemptLowerPriority
  priority: 2000001000
  priorityClassName: system-node-critical
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: kube-proxy
  serviceAccountName: kube-proxy
  terminationGracePeriodSeconds: 30
  tolerations:
  - operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/pid-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/network-unavailable
    operator: Exists
  volumes:
  - configMap:
      defaultMode: 420
      name: kube-proxy
    name: kube-proxy
  - hostPath:
      path: /run/xtables.lock
      type: FileOrCreate
    name: xtables-lock
  - hostPath:
      path: /lib/modules
      type: ""
    name: lib-modules
  - name: kube-api-access-m9xcs
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:49Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:49Z"
    message: 'containers with unready status: [kube-proxy]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:49Z"
    message: 'containers with unready status: [kube-proxy]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:49Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: containerd://d2b02ea8403433cc5e05c4e08de6f1310ae982b1bd8689299bb317c1c787ab21
    image: registry.k8s.io/kube-proxy:v1.26.3
    imageID: docker.io/library/import-2023-03-30@sha256:44db4d50a5f9c8efbac0d37ea974d1c0419a5928f90748d3d491a041a00c20b5
    lastState:
      terminated:
        containerID: containerd://d2b02ea8403433cc5e05c4e08de6f1310ae982b1bd8689299bb317c1c787ab21
        exitCode: 1
        finishedAt: "2023-04-19T20:09:28Z"
        reason: Error
        startedAt: "2023-04-19T20:09:28Z"
    name: kube-proxy
    ready: false
    restartCount: 6
    started: false
    state:
      waiting:
        message: back-off 5m0s restarting failed container=kube-proxy pod=kube-proxy-4pc46_kube-system(deed1d8c-fce3-494e-8e1f-52d7d0d2e819)
        reason: CrashLoopBackOff
  hostIP: 10.89.1.30
  phase: Running
  podIP: 10.89.1.30
  podIPs:
  - ip: 10.89.1.30
  qosClass: BestEffort
  startTime: "2023-04-19T20:03:49Z"
```
And coredns:
```
$ kubectl get pod coredns-787d4945fb-zhqtm -o yaml --namespace kube-system
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2023-04-19T20:03:49Z"
  generateName: coredns-787d4945fb-
  labels:
    k8s-app: kube-dns
    pod-template-hash: 787d4945fb
  name: coredns-787d4945fb-zhqtm
  namespace: kube-system
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: ReplicaSet
    name: coredns-787d4945fb
    uid: b5b6da79-7065-4b36-8353-073544ae5e79
  resourceVersion: "531"
  uid: 52618132-1b68-4922-b589-8cb555d32458
spec:
  affinity:
    podAntiAffinity:
      preferredDuringSchedulingIgnoredDuringExecution:
      - podAffinityTerm:
          labelSelector:
            matchExpressions:
            - key: k8s-app
              operator: In
              values:
              - kube-dns
          topologyKey: kubernetes.io/hostname
        weight: 100
  containers:
  - args:
    - -conf
    - /etc/coredns/Corefile
    image: registry.k8s.io/coredns/coredns:v1.9.3
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 5
      httpGet:
        path: /health
        port: 8080
        scheme: HTTP
      initialDelaySeconds: 60
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 5
    name: coredns
    ports:
    - containerPort: 53
      name: dns
      protocol: UDP
    - containerPort: 53
      name: dns-tcp
      protocol: TCP
    - containerPort: 9153
      name: metrics
      protocol: TCP
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /ready
        port: 8181
        scheme: HTTP
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    resources:
      limits:
        memory: 170Mi
      requests:
        cpu: 100m
        memory: 70Mi
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        add:
        - NET_BIND_SERVICE
        drop:
        - all
      readOnlyRootFilesystem: true
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /etc/coredns
      name: config-volume
      readOnly: true
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: kube-api-access-wdq2k
      readOnly: true
  dnsPolicy: Default
  enableServiceLinks: true
  nodeName: kind-cluster-control-plane
  nodeSelector:
    kubernetes.io/os: linux
  preemptionPolicy: PreemptLowerPriority
  priority: 2000000000
  priorityClassName: system-cluster-critical
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: coredns
  serviceAccountName: coredns
  terminationGracePeriodSeconds: 30
  tolerations:
  - key: CriticalAddonsOnly
    operator: Exists
  - effect: NoSchedule
    key: node-role.kubernetes.io/control-plane
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
    tolerationSeconds: 300
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
    tolerationSeconds: 300
  volumes:
  - configMap:
      defaultMode: 420
      items:
      - key: Corefile
        path: Corefile
      name: coredns
    name: config-volume
  - name: kube-api-access-wdq2k
    projected:
      defaultMode: 420
      sources:
      - serviceAccountToken:
          expirationSeconds: 3607
          path: token
      - configMap:
          items:
          - key: ca.crt
            path: ca.crt
          name: kube-root-ca.crt
      - downwardAPI:
          items:
          - fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
            path: namespace
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:53Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:53Z"
    message: 'containers with unready status: [coredns]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:53Z"
    message: 'containers with unready status: [coredns]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2023-04-19T20:03:53Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: containerd://daaa33dae2bea70c9d8818e90dade89a612c8629eb0312f8b40df487d7ba5df9
    image: registry.k8s.io/coredns/coredns:v1.9.3
    imageID: sha256:5185b96f0becf59032b8e3646e99f84d9655dff3ac9e2605e0dc77f9c441ae4a
    lastState: {}
    name: coredns
    ready: false
    restartCount: 0
    started: true
    state:
      running:
        startedAt: "2023-04-19T20:03:55Z"
  hostIP: 10.89.1.30
  phase: Running
  podIP: 10.244.0.4
  podIPs:
  - ip: 10.244.0.4
  qosClass: Burstable
  startTime: "2023-04-19T20:03:53Z"
```
Looking at events for `kube-system` namespace:
```
$ kubectl get events --namespace kube-system
LAST SEEN   TYPE      REASON              OBJECT                          MESSAGE
13m         Warning   FailedScheduling    pod/coredns-787d4945fb-2xclq    0/1 nodes are available: 1 node(s) had untolerated taint {node.kubernetes.io/not-ready: }. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..
13m         Normal    Scheduled           pod/coredns-787d4945fb-2xclq    Successfully assigned kube-system/coredns-787d4945fb-2xclq to kind-cluster-control-plane
13m         Normal    Pulled              pod/coredns-787d4945fb-2xclq    Container image "registry.k8s.io/coredns/coredns:v1.9.3" already present on machine
13m         Normal    Created             pod/coredns-787d4945fb-2xclq    Created container coredns
13m         Normal    Started             pod/coredns-787d4945fb-2xclq    Started container coredns
3m34s       Warning   Unhealthy           pod/coredns-787d4945fb-2xclq    Readiness probe failed: HTTP probe failed with statuscode: 503
13m         Warning   FailedScheduling    pod/coredns-787d4945fb-zhqtm    0/1 nodes are available: 1 node(s) had untolerated taint {node.kubernetes.io/not-ready: }. preemption: 0/1 nodes are available: 1 Preemption is not helpful for scheduling..
13m         Normal    Scheduled           pod/coredns-787d4945fb-zhqtm    Successfully assigned kube-system/coredns-787d4945fb-zhqtm to kind-cluster-control-plane
13m         Normal    Pulled              pod/coredns-787d4945fb-zhqtm    Container image "registry.k8s.io/coredns/coredns:v1.9.3" already present on machine
13m         Normal    Created             pod/coredns-787d4945fb-zhqtm    Created container coredns
13m         Normal    Started             pod/coredns-787d4945fb-zhqtm    Started container coredns
3m29s       Warning   Unhealthy           pod/coredns-787d4945fb-zhqtm    Readiness probe failed: HTTP probe failed with statuscode: 503
13m         Normal    SuccessfulCreate    replicaset/coredns-787d4945fb   Created pod: coredns-787d4945fb-2xclq
13m         Normal    SuccessfulCreate    replicaset/coredns-787d4945fb   Created pod: coredns-787d4945fb-zhqtm
13m         Normal    ScalingReplicaSet   deployment/coredns              Scaled up replica set coredns-787d4945fb to 2
13m         Normal    Scheduled           pod/kindnet-rpkvs               Successfully assigned kube-system/kindnet-rpkvs to kind-cluster-control-plane
13m         Normal    Pulling             pod/kindnet-rpkvs               Pulling image "docker.io/kindest/kindnetd:v20230330-48f316cd@sha256:c19d6362a6a928139820761475a38c24c0cf84d507b9ddf414a078cf627497af"
13m         Normal    Pulled              pod/kindnet-rpkvs               Successfully pulled image "docker.io/kindest/kindnetd:v20230330-48f316cd@sha256:c19d6362a6a928139820761475a38c24c0cf84d507b9ddf414a078cf627497af" in 1.686470464s (1.686476505s including waiting)
13m         Normal    Created             pod/kindnet-rpkvs               Created container kindnet-cni
13m         Normal    Started             pod/kindnet-rpkvs               Started container kindnet-cni
13m         Normal    SuccessfulCreate    daemonset/kindnet               Created pod: kindnet-rpkvs
13m         Normal    LeaderElection      lease/kube-controller-manager   kind-cluster-control-plane_870a3f8a-44d7-4431-8cda-e049b0233ba8 became leader
13m         Normal    Scheduled           pod/kube-proxy-4pc46            Successfully assigned kube-system/kube-proxy-4pc46 to kind-cluster-control-plane
12m         Normal    Pulled              pod/kube-proxy-4pc46            Container image "registry.k8s.io/kube-proxy:v1.26.3" already present on machine
12m         Normal    Created             pod/kube-proxy-4pc46            Created container kube-proxy
12m         Normal    Started             pod/kube-proxy-4pc46            Started container kube-proxy
3m29s       Warning   BackOff             pod/kube-proxy-4pc46            Back-off restarting failed container kube-proxy in pod kube-proxy-4pc46_kube-system(deed1d8c-fce3-494e-8e1f-52d7d0d2e819)
13m         Normal    SuccessfulCreate    daemonset/kube-proxy            Created pod: kube-proxy-4pc46
13m         Normal    LeaderElection      lease/kube-scheduler            kind-cluster-control-plane_4b812fe8-d1e9-4524-badf-3d4e30cf7ca1 became leader

```

**Comment by odockal** [comment]
**Linux resolution:**
I had to restart my machine, saw some kernel-core error on the way. I tried again and now I can nicely spawn kind cluster with all pods/containers running in it. 
then I deployed (what is the right terminology) the httpdbin (https://projectcontour.io/examples/httpbin.yaml) yaml via PLay Kubernetes yaml. App was not accessible on `localhost:9090`. Looking at pod's logs I saw `too many files` error. So I have updated the deployment to use just one replica and voila! It is working now.

although I am curious how can I affect kube cluster system setup so that I can change limits set in `ulimit` in containers in general.
